### PR TITLE
docs: update Readme.md

### PR DESCRIPTION
**Pull request by yuxizama**
- Reorganized existing content to follow the new structure
- Aligned all the header links and badges
- Added a table of contents
- Added a call-to-action for users to star the repo
- Checked all links and replaced outdated ones.
- Unified the formatting
- Updated the support banner
- Improved the overall content quality
- Added missing content where necessary
- Added more “resources” for each product
- Ensured wording is consistent throughout

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,55 +1,92 @@
...
+
+Thanks to a breakthrough in homomorphic encryption, Zama’s fhEVM makes it possible to run confidential smart contracts on encrypted data, guaranteeing both confidentiality and composability with:
 - **End-to-end encryption of transactions and state:** Data included in transactions is encrypted and never visible to anyone.
 - **Composability and data availability on-chain:** States are updated while remaining encrypted at all times.
-- **No impact on existing dapps and state:** Encrypted state co-exist alongside public one, and doesn't impact existing dapps.
+- **No impact on existing dapps and state:** Encrypted state co-exists alongside public one, and doesn't impact existing dapps.
+<br></br>
 
-### Developers can write confidential smart contracts without learning cryptography
+### Main features
```
Comment (yuxizama) [inline README.md:30]: Is everything correct and up to date in the "main features" 
Also, we would like to only include main features here, so if there are some minor ones or less important ones, we can consider to remove it.

File: README.md [hunk 2]
```diff
@@ -85,25 +127,72 @@ contract Counter {
...
+### Development Guide
 
 Install dependencies (Solidity libraries and dev tools)
 
 ```bash
 npm install
 ```
 
-Note: Solidity files are formatted with prettier.
+> ℹ️ Note: Solidity files are formatted with prettier.
```
Comment (aquint-zama) [inline README.md]: you could use Github admonition

> [!NOTE]
> this is a note

File: README.md [hunk 3]
```diff
@@ -85,25 +127,72 @@ contract Counter {
...
+### Documentation
+
+Full, comprehensive documentation is available here: [https://docs.zama.ai/concrete-ml](https://docs.zama.ai/fhevm).
+<p align="right">
+  <a href="#table-of-contents" > ↑ Back to top </a> 
+</p>
+
+## Working with fhEVM
+
+### Development Guide
```
Comment (yuxizama) [inline README.md]: The development guide" is for contributing to the project ? 
We also have another section "contributing"  below and it says: 
```
There are two ways to contribute to the Zama fhEVM:
- [Open issues](https://github.com/zama-ai/fhevm/issues/new/choose) to report bugs and typos, or to suggest new ideas
- Request to become an official contributor by emailing hello@zama.ai.
```
Are they in conflict?

File: README.md [hunk 4]
```diff
@@ -85,25 +127,72 @@ contract Counter {
...
 
-### Generate TFHE lib
+#### Generate TFHE lib
 
 ```
 npm run codegen
 ```
 
-WARNING: Use this command to generate Solidity code and prettier result automatically!
+> ⚠️ WARNING: Use this command to generate Solidity code and prettier result automatically!
```
Comment (aquint-zama) [inline README.md]: you could use Github admonition

> [!WARNING]
> this is a warning

File: README.md [hunk 5]
```diff
@@ -212,27 +304,63 @@ Then open the file `coverage/index.html`. You can see there which line or branch
...
 
-- you can open issues to report bugs or typos, or to suggest new ideas
-- you can ask to become an official contributor by emailing hello@zama.ai. (becoming an approved contributor involves signing our Contributor License Agreement (CLA))
-  Only approved contributors can send pull requests, so please make sure to get in touch before you do!
+- [Open issues](https://github.com/zama-ai/fhevm/issues/new/choose) to report bugs and typos, or to suggest new ideas
+- Request to become an official contributor by emailing hello@zama.ai.
 
-## Credits
+Becoming an approved contributor involves signing our Contributor License Agreement (CLA)). Only approved contributors can send pull requests, so please make sure to get in touch before you do!
+<br></br>
```
Comment (yuxizama) [inline README.md:344]: Here the "Contributing" part

**Review by aquint-zama — APPROVED** [review]
2 minor suggestion that you could feel free to ignore, otherwise looks good to me!

**Review by yuxizama — COMMENTED** [review]
Hey! 
Here are some questions left before finalizing the Readme pages. 
Thanks!

File: README.md (hunk 3 above)
Comment (aquint-zama) [reply README.md]: No they aren't, it's two different ways:
- giving feedbacks/asking for features 
or
- contributing: sharing code for fixes/new features

Comment (aquint-zama) [reply README.md]: development guide, is more Developer/User guide

File: README.md [hunk 6]
```diff
@@ -1,55 +1,92 @@
...
 - **Simple Developer Experience:** Developers can use the `euint` data types to mark which part of their contracts should be private.
 - **Programmable Privacy:** All the logic for access control of encrypted states is defined by developers in their smart contracts.
+- **High Precision Encrypted Integers :** Up to 256 bits of precision for integers
+- **Full range of Operators :** All typical operators are available: `+`, `-`, `*`, `/`, `<`, `>`, `==`, …
+- **Encrypted If-Else Conditionals :** Check conditions on encrypted states
+- **On-chain PRNG :** Generate secure randomness without using oracles
+- **Configurable Decryption :** Threshold, centralized or KMS decryption
+- **Unbounded Compute Depth :** Unlimited consecutive FHE operations
+
+*Learn more about Concrete ML features in the [documentation](https://docs.zama.ai/fhevm).*
```
Comment (aquint-zama) [inline README.md]: about **fhEVM** features

File: README.md [hunk 7]
```diff
@@ -85,25 +127,72 @@ contract Counter {
...
+- [Confidential ERC-20 Tokens Using Homomorphic Encryption and the fhEVM](https://www.zama.ai/post/confidential-erc-20-tokens-using-homomorphic-encryption)
+- [On-chain Blind Auctions Using Homomorphic Encryption and the fhEVM](https://www.zama.ai/post/on-chain-blind-auctions-using-homomorphic-encryption)
+- [Programmable Privacy and Onchain Compliance using Homomorphic Encryption](https://www.zama.ai/post/programmable-privacy-and-onchain-compliance-using-homomorphic-encryption)
+  
+*Explore more useful resources in [fhEVM tutorials](https://docs.zama.ai/fhevm/resources/tutorials) and [Awesome Zama repo](https://github.com/zama-ai/awesome-zama)*
+<br></br>
+
+### Documentation
+
+Full, comprehensive documentation is available here: [https://docs.zama.ai/concrete-ml](https://docs.zama.ai/fhevm).
```
Comment (aquint-zama) [inline README.md]: link is CML one

**Review by aquint-zama — CHANGES_REQUESTED** [review]
some CML refs to update

File: README.md (hunk 3 above)
Comment (yuxizama) [reply README.md]: I see!

**Comment by yuxizama** [comment]
I've made some last few commits. 
May you check the change? @aquint-zama 
If all good we can merge this PR. 
Thanks!

**Review by aquint-zama — APPROVED** [review]

**Pull request merged**